gui: add max button for spend recipients

jp1ac4 opened the pull request:
This adds the MAX button from #546.

The max amount is calculated within the `redraft` function based on the change output. The recipient's amount is updated directly within this same function in order to avoid another call to `redraft` after updating the amount.

The MAX button only takes effect once when the user clicks it. If further changes are made (e.g. to feerate or other recipients), the value will not update and the user will need to click MAX again.

Currently, the MAX button is always clickable. It would be possible to determine for each recipient whether the MAX button should be clickable for that recipient (feerate is valid and all other recipients are valid and no duplicates), but it's not a trivial change so I'm not sure if it's worth doing as I think the behaviour is quite intuitive.

pythcoiner commented:
Love it!

What about greying out `Max` button if no fee rate/no coin selected?

pythcoiner commented:
i don't understand how it can be related to this PR, but i've one issue that i don't have on master:

![image](https://github.com/wizardsardine/liana/assets/124568858/72a3733c-ac12-40fa-bf22-c1c7e1c0f5d7)

i'm creating a new signet wallet, when i import key from coldcard, i got an `Xpub` instead a `Tpub`

edouardparis commented:
Working great, could it be possible to put the button it in a tooltip which on hover would display:
`Quickly calculate the remaining amount` or something like this.

jp1ac4 commented:
In the end, it was simpler than I thought to add an exact condition on whether the max button for a given recipient should be clickable (it might be clickable for one recipient and not another).

I've also added a tooltip as suggested.

pythcoiner commented:
ACK  4e252d4

edouardparis commented:
>In the end, it was simpler than I thought to add an exact condition on whether the max button for a given recipient should be clickable (it might be clickable for one recipient and not another).

I cannot deactivate it to use it to another recipient.
If we lock the button, then user should be able to unlock it. 
Either:
1. we let it be a clickable button that redo calculation on click,
2. or we make it a "toggle" button: 
  - border became green to say it is active, user can deactivate it, only one recipient hast it.
  - and redraft takes it into account to redo calculation and modify the recipient amount at each modification of feerate.
  - The recipient amount form is greyed out as long as the Max button is active.

jp1ac4 commented:
I've changed the button to a checkbox that remains selected and continues to recalculate the max for that recipient as changes are made elsewhere.

The checkbox can be selected at any time, but the max amount will only be calculated as long as the selected recipient's address is valid, all other recipients are valid and the feerate is valid. If the user has selected coins manually, then at least one coin must be selected and otherwise all coins will be selected for the calculation.

jp1ac4 commented:
Thanks, @edouardparis, for your testing feedback.

I've made some further changes (some in separate commits as they are not directly related to the max button):
- Add alternative message to "Feerate needs to be set" in case something else is missing.
- If there is not enough information to redraft (incomplete form values), set the recipient's max amount to empty string and unset the amount left to select.
- if form values are valid and no coins are selected and it's user selection, set max amount to 0.
- if the amount left to select is 0, we only display "0 left to select" if the "Next" button is clickable, and otherwise display a message indicating what's still missing.

edouardparis commented:
It is working great ! Good work !

edouardparis commented:
Sorry a little nit, but I am ok to merge the PR.
It feels like the tooltip `Quickly calculate the remaining amount`does not provide a good explanation.
What do you think of something like: `Total amount of the selected coins less the other recipients amount and fees` ?

jp1ac4 commented:
I think changing the wording makes sense given that now the checkbox remains selected.

Perhaps `Total amount remaining from selected coins after paying fee and any other recipients` ?

It might be better to omit "from selected coins" as it's not completely accurate in the case of auto-selection, so `Total amount remaining after paying fee and any other recipients`.

I've used "fee" rather than "fees" but I think both could work.

edouardparis commented:
> Total amount remaining after paying fee and any other recipients

Ok for this.

jp1ac4 commented:
I've updated the tooltip wording and also changed its position to `Left` as otherwise it was hard to see the final letters.

Updated version:
![image](https://github.com/wizardsardine/liana/assets/121959000/c2837e2c-8585-476f-ae93-848f4094b097)

edouardparis reviewed (approved):
ACK 46cd0f4923c5fe59a5ff08134c850d9a9ceb345a
